pygac/reader.py

```diff
@@ -513,68 +515,123 @@ def get_sun_earth_distance_correction(self):
         return calculate_sun_earth_distance_correction(jday)
 
     def update_meta_data(self):
-        """Add some metd data to the meta_data dicitonary."""
-        if 'sun_earth_distance_correction_factor' not in self.meta_data:
-            self.meta_data['sun_earth_distance_correction_factor'] = (
+        """Add some meta data to the meta_data dicitonary."""
+        meta_data = self.meta_data
+        self._update_meta_data_object(meta_data)
+        if 'gac_header' not in meta_data:
+            meta_data['gac_header'] = self.head.copy()
+
+    def _update_meta_data_object(self, meta_data):
+        if 'sun_earth_distance_correction_factor' not in meta_data:
+            meta_data['sun_earth_distance_correction_factor'] = (
                 self.get_sun_earth_distance_correction())
-        if 'midnight_scanline' not in self.meta_data:
-            self.meta_data['midnight_scanline'] = self.get_midnight_scanline()
-        if 'missing_scanlines' not in self.meta_data:
-            self.meta_data['missing_scanlines'] = self.get_miss_lines()
-        if 'gac_header' not in self.meta_data:
-            self.meta_data['gac_header'] = self.head.copy()
-        self.meta_data['calib_coeffs_version'] = self.calibration.version
+        if 'midnight_scanline' not in meta_data:
+            meta_data['midnight_scanline'] = self.get_midnight_scanline()
+        if 'missing_scanlines' not in meta_data:
+            meta_data['missing_scanlines'] = self.get_miss_lines()
+
+    def read_as_dataset(self, file_to_read):
+        self.read(file_to_read)
+        return self.create_counts_dataset()
+
+    def create_counts_dataset(self):
```

Comment:
Is it possible to split that method into smaller parts?

Reply:
Done